### PR TITLE
Add documentation for the ProduceOnlyReferenceAssembly msbuild option

### PR DESCRIPTION
**Pull request by chsienki**
Documents the ProduceOnlyReferenceAssembly option in the refout docs

### CODE REVIEW

**Comment by chsienki** [comment]
I've also opened an issue for on docs (https://github.com/MicrosoftDocs/visualstudio-docs/issues/1407) to get it added to the common msbuild project properties section when it becomes available.

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review.

File: docs/features/refout.md [hunk 1]
```diff
@@ -83,7 +83,8 @@ That said, even if determinism isn't set, compilation of ref assemblies is [larg
 
 ## MSBuild
 
-* `ProduceReferenceAssembly` (boolean) controls whether to create the item passed to the compiler task (and thus pass `/refout:`). It requires opt-in. It is recommended that `Deterministic` also be set for best result (see details above).
+* `ProduceReferenceAssembly` (boolean) controls whether to create the item passed to the compiler task (and thus pass `/refout:`). It requires opt-in. It is recommended that `Deterministic` also be set for best result (see details above).  Cannot be used in conjunction with `ProduceOnlyReferenceAssembly`
+* `ProduceOnlyReferenceAssembly` (boolean) controls whether to pass `/refonly` to the compiler. It requires opt-in. Cannot be used in conjunction with `ProduceReferenceAssembly`
```
Comment (jaredpar) [inline docs/features/refout.md]: Nit: inconsistency on whether or not you put a colon after the option name.

Comment (chsienki) [reply docs/features/refout.md]: `/refout:` takes a path as an option. `/refonly` is just a switch with nothing passed to it.

Comment (jaredpar) [reply docs/features/refout.md]: Ah yes

**Pull request merged**